### PR TITLE
Fixed error: pcap library not found! in configure.ac file for FreeBSD

### PR DESCRIPTION
**Pull request by Martinfx**
Fixed error: pcap library not found for FreeBSD
```
[xxx@ /reaver-wps-fork-t6x/src]$ ./configure 
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
./configure: FreeBSD: not found
checking for FreeBSD... checking for pcap_open_live in -lpcap... no
error: pcap library not found!
exit: Illegal number: -1
```

### CODE REVIEW

File: src/configure.ac [hunk 1]
```diff
@@ -3,8 +3,18 @@ AC_INIT(reaver, 1.5)
 AC_PROG_CC
 AC_LANG(C)
 
+
 CFLAGS="-Wall $CFLAGS"
-LDFLAGS="-ldl -lm -lpcap -lsqlite3 $LDFLAGS"
+
+
+if test `uname` = "FreeBSD"; then
+  LDFLAGS="-lpcap -lsqlite3 $LDFLAGS"
```
Comment (rofl0r) [inline src/configure.ac]: i think `-ldl` is not even needed for linux. just removing it would be much cleaner than this ifdef hell.

File: src/configure [hunk 2]
```diff
@@ -577,8 +577,8 @@ MAKEFLAGS=
 # Identity of this package.
 PACKAGE_NAME='reaver'
 PACKAGE_TARNAME='reaver'
-PACKAGE_VERSION='1.5.3'
-PACKAGE_STRING='reaver 1.5.3'
+PACKAGE_VERSION='1.5'
```
Comment (rofl0r) [inline src/configure]: why did you change the version ?

File: src/lwe/iwlib.h [hunk 3]
```diff
@@ -27,10 +27,31 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <netdb.h>		/* gethostbyname, getnetbyname */
+#include <netdb.h>		    /* gethostbyname, getnetbyname */
```
Comment (rofl0r) [inline src/lwe/iwlib.h]: i'm not a fan of gratuitous whitespace changes

File: src/lwe/iwlib.h [hunk 4]
```diff
@@ -27,10 +27,31 @@
...
 #include <net/ethernet.h>	/* struct ether_addr */
 #include <sys/time.h>		/* struct timeval */
 #include <unistd.h>
+#include <stdint.h>
+
+/*
+ * This are types for compatibily
+ */
+#ifdef __linux__
+  #include "linux/types.h"
```
Comment (rofl0r) [inline src/lwe/iwlib.h]: maybe this can be entirely removed.
if you can compile that despite not having a header included that defines caddr_t (like mentioned in the original comment, that you removed) that's a strong hint in that direction.

File: src/lwe/iwlib.h [hunk 5]
```diff
@@ -27,10 +27,31 @@
...
+  #include <net/if_arp.h>		 /* For ARPHRD_ETHER */
+#else
+
+typedef unsigned char  u_char;
```
Comment (rofl0r) [inline src/lwe/iwlib.h]: this stuff shouldnt be necessary. it's defined in sys/types.h when _GNU_SOURCE is defined.
see https://github.com/t6x/reaver-wps-fork-t6x/pull/89/commits/bd357b580de8cbaf1f3cf008fbd64de9964ffd63

**Comment by rofl0r** [comment]
btw: that you get an error about pcap not found if you installed it into /usr/local is not an error.
/usr/local/lib and include/ are not in the default compiler path, so the onus is on you to supply the correct -L and -I paths via CFLAGS.

File: src/lwe/iwlib.h (hunk 5 above)
Comment (Martinfx) [inline src/lwe/iwlib.h]: Its true. Sorry for duplicate code

File: src/lwe/iwlib.h (hunk 4 above)
Comment (Martinfx) [inline src/lwe/iwlib.h]: Yes linux headers was problem.

**Comment by rofl0r** [comment]
i addressed most of these issues here in #99 and #98

File: src/configure (hunk 2 above)
Comment (Martinfx) [inline src/configure]: I dont know, i pull from repository, maybe i am mistake changed. Sorry

**Comment by Martinfx** [comment]
Thanks for review code

**Comment by kcdtv** [comment]
Is it OK now for you with the actual main version?
I just pushed 3 commits full of fixes (thanks rofl0r :smiley_cat:)? 
Can you try to see?
Many thanks for your contribution Martinfx

**Comment by Martinfx** [comment]
configure is now fixed (rofl0r send patch). Maybe last my commit is good.

**Comment by kcdtv** [comment]
If i understand you well rofl0r sent you some patches and the last version of your contribution is giving satisfactory results,. 
Your commit has some conflicts with the branch, do you mind to propose a new request using the main branch as it is now? 
That will be simpler for me. 
Take care and thanks to both of you.

**Comment by Martinfx** [comment]
i close this pull request and send new.
